From f25a5383cad353df5ba95bcf4fa0b21e84e228ea Mon Sep 17 00:00:00 2001
From: sahnalys11 <vincent.vangestel_ext@softathome.com>
Date: Wed, 7 Oct 2020 10:43:41 +0200
Subject: [PATCH] use regular getenv instead of secure_getenv

---
 src/amxo_import_resolver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/amxo_import_resolver.c b/src/amxo_import_resolver.c
index ca6b46a..266499a 100644
--- a/src/amxo_import_resolver.c
+++ b/src/amxo_import_resolver.c
@@ -100,7 +100,7 @@ static void amxo_import_lib_free(AMXO_UNUSED const char* key,
                                  amxc_htable_it_t* it) {
     amxo_import_lib_t* import =
         amxc_htable_it_get_data(it, amxo_import_lib_t, hit);
-    char* no_dlclose = secure_getenv("AMXO_NO_DLCLOSE");
+    char* no_dlclose = getenv("AMXO_NO_DLCLOSE");
 
     if(no_dlclose == NULL) {
         dlerror();
